go.mod

```diff
@@ -344,7 +344,7 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/prometheus/prometheus => github.com/grafana/mimir-prometheus v1.8.2-0.20250617052252-226f974e041b
+replace github.com/prometheus/prometheus => github.com/grafana/mimir-prometheus v1.8.2-0.20250619151923-05b1e6ec6c46
```

Comment:
Note to reviewer: I am pointing this to https://github.com/grafana/mimir-prometheus/tree/lamida/bytes-string-labels while waiting https://github.com/prometheus/prometheus/pull/16717 get approved.

Reply:
Are you hoping to merge this PR before that Prometheus PR is accepted?

If so, we'll need to merge your change into mimir-prometheus `main` (otherwise it'll be lost next time we sync mimir-prometheus with upstream and bring that into Mimir).

Reply:
I am fine for waiting this PR to be merged after prometheus PR is merged, unless if you think we should do before that.

Reply:
@charleskorn on the second thought, I think let's merge the promethes change to mimir-prometheus first. I created this PR: https://github.com/grafana/mimir-prometheus/pull/908. Let me know if that is not the right way to do that.

Reply:
Let's give the Prometheus PR another day or two, it'll save us a bit of faffing about if it's merged soon.